### ISSUE TITLE
[🐞] CLI - Build script silences failed steps when running with Yarn

### ISSUE THREAD

**Issue by ianlet** [QwikDev/qwik#5104]
### Which component is affected?

Starters / CLI

### Describe the bug

If you use Yarn, running the `yarn build` command while there are (for example) linting issues will prematurely exit the process without printing anything about why it failed.

### Reproduction

https://stackblitz.com/edit/qwik-starter-pcpq7a?file=src%2Froot.tsx

### Steps to reproduce

* Make a compiling but invalid change in a file (e.g. declare an unused variable in `src/root.tsx`)
* Run `yarn build`
* The command will fail without printing any valuable information on what failed

Here's an example:
```
$ yarn build
      ............
    .::: :--------:.
   .::::  .:-------:.
  .:::::.   .:-------.
  ::::::.     .:------.
 ::::::.        :-----:
 ::::::.       .:-----.
  :::::::.     .-----.
   ::::::::..   ---:.
    .:::::::::. :-:.
     ..::::::::::::
             ...::::
     


yarn run build.types
yarn run build.client
yarn run lint

$ vite build
vite v4.4.7 building for production...
✓ 95 modules transformed.
dist/build/q-8f7ef15c.webp   3.96 kB
dist/build/q-da3d1958.webp   7.49 kB
dist/build/q-85611e8a.webp  11.42 kB
dist/build/q-e86984e2.webp  15.85 kB
dist/build/q-7df0c9e1.webp  27.56 kB
dist/q-manifest.json        30.97 kB │ gzip:  5.41 kB
dist/build/q-027f67b0.css    4.00 kB │ gzip:  1.28 kB
dist/build/q-86630e49.js     0.13 kB │ gzip:  0.12 kB
dist/build/q-ff9c7aea.js     0.13 kB │ gzip:  0.13 kB
dist/build/q-dcd6eea6.js     0.21 kB │ gzip:  0.16 kB
dist/build/q-41dd0fb6.js     0.24 kB │ gzip:  0.19 kB
dist/build/q-c37f9393.js     0.26 kB │ gzip:  0.21 kB
dist/build/q-d0ad1337.js     0.31 kB │ gzip:  0.23 kB
dist/build/q-0b186ed1.js     0.38 kB │ gzip:  0.28 kB
dist/build/q-ffc9fb89.js     0.41 kB │ gzip:  0.29 kB
dist/build/q-51da6c56.js     0.47 kB │ gzip:  0.32 kB
dist/build/q-b2942bd8.js     0.54 kB │ gzip:  0.32 kB
dist/build/q-599a7cd5.js     0.55 kB │ gzip:  0.35 kB
dist/build/q-22949cf9.js     0.70 kB │ gzip:  0.44 kB
dist/build/q-4fd745e3.js     0.75 kB │ gzip:  0.45 kB
dist/build/q-3be6cc27.js     0.77 kB │ gzip:  0.41 kB
dist/build/q-c2a8b70a.js     0.79 kB │ gzip:  0.51 kB
dist/build/q-6022aabb.js     0.89 kB │ gzip:  0.55 kB
dist/build/q-cb96be76.js     0.96 kB │ gzip:  0.51 kB
dist/build/q-a8001a11.js     1.04 kB │ gzip:  0.65 kB
dist/build/q-25e27e0c.js     1.14 kB │ gzip:  0.67 kB
dist/build/q-3eab91df.js     1.15 kB │ gzip:  0.64 kB
dist/build/q-856e3d49.js     1.17 kB │ gzip:  0.58 kB
dist/build/q-8b587fdb.js     1.20 kB │ gzip:  0.62 kB
dist/build/q-44d16505.js     1.50 kB │ gzip:  0.73 kB
dist/build/q-3262ec8a.js     1.67 kB │ gzip:  0.89 kB
dist/build/q-663033b0.js     2.29 kB │ gzip:  1.11 kB
dist/service-worker.js       2.54 kB │ gzip:  1.21 kB
dist/build/q-de896cde.js     2.69 kB │ gzip:  1.23 kB
dist/build/q-0db0dd48.js     2.83 kB │ gzip:  0.95 kB
dist/build/q-cb2d40cb.js     3.82 kB │ gzip:  1.83 kB
dist/build/q-13e363e3.js     5.40 kB │ gzip:  2.14 kB
dist/build/q-f2a74367.js     5.85 kB │ gzip:  2.45 kB
dist/build/q-070b48f1.js     8.75 kB │ gzip:  4.13 kB
dist/build/q-1e8e1aad.js    49.24 kB │ gzip: 20.00 kB
✓ built in 3.63s

✓ Built client modules

error Command failed with exit code 1.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### System Info

```shell
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 16.20.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.4.2 - /usr/local/bin/npm
    pnpm: 8.6.10 - /usr/local/bin/pnpm
  npmPackages:
    @builder.io/qwik: ^1.2.10 => 1.2.10 
    @builder.io/qwik-city: ^1.2.10 => 1.2.10 
    undici: 5.22.1 => 5.22.1 
    vite: 4.4.7 => 4.4.7
```

### Additional Information

I tried with more recent versions of Node (v20.5.1) and Yarn (v4.0.0-rc.50), but it's the same.

**Comment by maciej-ka** [comment]
Hi, I'm looking into it

**Comment by maciej-ka** [comment]
Are these warnings expected when using `pnpm install`?
My dev setup seems to fail as a consequence of them.

<img width="1156" alt="Screenshot 2023-09-27 at 01 49 02" src="https://github.com/BuilderIO/qwik/assets/5403694/2798fcfe-9b07-49ff-81f8-d06706147325">
<img width="1182" alt="Screenshot 2023-09-27 at 01 49 28" src="https://github.com/BuilderIO/qwik/assets/5403694/c9b64522-fd06-4ace-b79c-121bef7ff982">

**Comment by maciej-ka** [comment]
_I don't know why I have this old `pnpm` problem that I cannot pass._

For anyone who will continue this ticket:

The file running behind `qwik build` is:
`/packages/qwik/src/cli/build/run-build-command.ts`

The first step could be to debug/console log silent several catch clauses:
```js
    }).catch(() => {
      process.exit(1);
    });
```

My plan to create a dev setup was:
```bash
cd ~
git clone git@<your-fork-here>/qwik.git
cd qwik
pnpm install
pnpm build
pnpm link.dist
```

```bash
cd ~
pnpm create qwik
cd qwik-app
pnpm install
pnpm link --global @builder.io/qwik @builder.io/qwik-city
pnpm build
```

I wish you all the best!
And I hope we meet on another occasion.

**Comment by Charlygraphy23** [comment]
let me check if i find something, i will share the same

**Comment by Charlygraphy23** [comment]
Thanks for the reproduction link - 

>Reproduction
https://stackblitz.com/edit/qwik-starter-pcpq7a?file=src%2Froot.tsx

First i try `yarn build` command and i am able to reproduce the issue, the issue with the yarn is, its not showing the proper description from where the error occurs. 
Then i tried `npm run build` and i found there was an unused variable that causing the build to failed if you remove the following it should work properly 

```diff

import './global.css';

+const iAmUnused = '';
-const iAmUnused = ''; 

```
So now `yarn build` working fine.

But here new issue arises, in the time `yarn build` why does the exact error not get printed but in the time of `npm build` the error get printed in the console?

**Comment by ianlet** [comment]
@Charlygraphy23 this is what this issue is actually about. As stated by @maciej-ka in [this message](https://github.com/BuilderIO/qwik/issues/5104#issuecomment-1737186419), it's probably caused by the errors being silenced in different catch clauses.

I could have a look at it later this week, but if anyone else wants to tackle this in the meantime (should be an easy fix), go ahead.

**Comment by gioboa** [comment]
Closed by #5267
